src/xpk/core/docker_manager.py

```diff
@@ -14,39 +14,98 @@
 limitations under the License.
 """
 
+from abc import ABC, abstractmethod
 import docker
-from ..utils.console import xpk_print
-from docker.errors import ImageNotFound
-from shutil import move
+from docker.errors import ContainerError, APIError, ImageNotFound, BuildError
+from ..utils.console import xpk_print, xpk_exit
+from shutil import copytree, copy
+import requests
+import os
+import tempfile
 
+
+DockerRunCommandExitCode = 135
+dockerBuildErrorCode = 134
 ctk_dockerfile_path = "Dockerfile"
 ctk_docker_image = "xpk-ctk"
-gcloud_cfg_mount_path = "/gcloud_cfg"
-deployment_dir_mount_path = "/deployment"
-xpk_ctk_img_name = "gcluster-xpk"
+ctk_container_name = "xpk-ctk-container"
+gcloud_cfg_mount_path = "/root/.config/gcloud"
+working_dir_mount_path = "/out"
+dockerfile_gh_path = "https://raw.githubusercontent.com/GoogleCloudPlatform/cluster-toolkit/refs/heads/develop/tools/cloud-build/images/cluster-toolkit-dockerfile/Dockerfile"
+
+
+class CommandRunner(ABC):
+  """This is a base class that defines methods a class for running cluster toolkit command should implement."""
+
+  @abstractmethod
+  def build(self) -> None:
```

Comment:
build is part of initialize() now and no need to expose to outside world.

Initialize():
currently: downloads dockerfile and builds an image
in the future: will pull the image from a registry (no build)

Reply:
done